servermaster/scheduler/scheduler.go

```diff
@@ -71,7 +71,11 @@ func (s *Scheduler) checkCostAllows(
 	request *schedModel.SchedulerRequest,
 	target model.ExecutorID,
 ) bool {
-	executorResc := s.capacityProvider.CapacityForExecutor(target)
+	executorResc, ok := s.capacityProvider.CapacityForExecutor(target)
```

Comment:
The `CapacityForExecutor` or `CapacitiesForAllExecutors` returns a capacity snapshot, if the `scheduler.ScheduleTask` is called concurrently, the capacity usage could have write skew, such as available capacity is 100, and two concurrent tasks asks for 80 cap each other, they are both allowed to start.

Is there any concurrency control to avoid such problem.

Reply:
Added comments on `CapacityProvider`.